### ISSUE TITLE
Non existing folders appear on top

### ISSUE THREAD

**Issue by lesauxvi** [aiekick/ImGuiFileDialog#189]
I've an issue that appear only on Windows apparently (I've not seen this issue on MacOS, nor Linux) where non existing folders appear on top  (see Figure below). In this example, there are three folders (two start with `!!!!!`, the other one with `$`).

<img width="496" alt="Capture d’écran 2024-07-08 à 10 50 59" src="https://github.com/aiekick/ImGuiFileDialog/assets/8177167/e85f6ee0-e30b-4a09-9fb9-4246d519ce9f">

If I double click on one  of these folder, I get an empty directory.

The configuration I use is here:
```
IGFD::FileDialogConfig config;
config.path = workspace.default_directory(); // specific to my application
constexpr auto IGFD_FLAGS = ImGuiFileDialogFlags_Modal | ImGuiFileDialogFlags_DontShowHiddenFiles | ImGuiFileDialogFlags_NaturalSorting;
config.flags = IGFD_FLAGS;
config.countSelectionMax = 0;
const auto filters = "Infrared files (*.irb,*.nvf,*.ptw){.irb,.nvf,.ptw}";
ImGuiFileDialog::Instance()->OpenDialog("import_ir_files_dialog", ICON_MS_WEB_STORIES "  Select the infrared files to import", filters, config);
```
and my configuration file is  [here](https://github.com/user-attachments/files/16125158/ImGuiFileDialogConfig.h.txt). I had to add `.txt` to be able to upload the file. This appear on local drives, as well as on network drives.

Are you aware of this issue?

**Comment by aiekick** [comment]
Hello,

thanks for the report.

not im not awarea about rhis bug, and its very weird. never got a similar bug..

are you sure these directories are not in your fs ? maybe hidden or something ?

**Comment by aiekick** [comment]
I have checked on my side, and cant found a bug. 
Can you debug IGFD, put a breakpoint in the ScanDirectory func of the fs interface you are using.
check files by file. the goal is to see if your fs list theses files or not ?

for me these dir must exist on your fs, if not i not see how its possible to enter in them with IGFD.
since the existence of the path is tested before enter it even they are finally empty.

**Comment by lesauxvi** [comment]
I used the debug mode and placed a breakpoint where you suggest me to do. Some non existing files are seen (see below for one example). They do not appear on the Windows explorer, even with the "show hidden files and folders" checked). That is really weird.

<img width="1223" alt="Capture d’écran 2024-07-09 à 19 21 00" src="https://github.com/aiekick/ImGuiFileDialog/assets/8177167/644ab2b6-a141-4fc3-99bc-6a4654155d69">

<img width="1223" alt="Capture d’écran 2024-07-09 à 19 22 11" src="https://github.com/aiekick/ImGuiFileDialog/assets/8177167/71ed635a-4c3e-4af6-a96d-47414c54f724">

I will check with my IT tomorrow if they are aware of these hidden files (I'm not administrator in this windows machine, so I do not master everything). 

I'm coming back to you tomorrow.

**Comment by aiekick** [comment]
yes its very weird

one time i guess could be some buffer overflow so with a weird name, but here the ext is good. but if your fs directly list theses files so they are existing somewhere.

 i know ntfs is a powerful fs who can hide files in many ways. so a maybe a lost tmp file , or a virus who saved datas in theses files and have a way for hide them from your file explorer. 

yes your admins can have an idea or inspect them.

**Comment by lesauxvi** [comment]
I did some experiments this morning on several Windows machine that can be classified into two categories: machines configured by our IT and personal laptops. It appears that the issue appears only on the machines managed by our IT. 

After a discussion with IT guys,  these strange folders and files are actually created by our antivirus software in order to catch malicious softwares (see [here](https://forums.ivanti.com/s/article/A-series-of-random-named-files-with-exclamation-marks-appear-on-a-File-Director-Map-Point?language=en_US) for a more detailed explanation).

The syntax of these hidden files/folders seem to be imposed: 5 ! followed by 10 digits (and sometimes an extension). I suggest two solutions:

1. add a small paragraph somewhere to inform the users that this may happen depending on the machine configuration
2. add an exception rule in the ScanDirectory function to avoid the files that respect the naming convention (5 ! followed by 10 digits)

What do you think?

PS : the problem occurs using the std::filesystem interface, as well as dirent.

**Comment by aiekick** [comment]
> PS : the problem occurs using the std::filesystem interface, as well as dirent.

To be clear, this is not a bug or a problem, IGFD does what it is supposed to do : list files from the file system

In a second time if a user want avoid the listing of some patterns due to his side configuration, is an another thing.

So now you can use regex for maybe avoid these files if you have a pattern who correpond to your files except theses one, or override the fs interface for remove some specific files

**Comment by lesauxvi** [comment]
Yes, IGFD does exactly what it is supposed to do. So, I suggest to close this issue as it is not related to the library.

The best thing for my case is what you suggest (or if I'm lazy just to inform the user of the "issue").

Would it make sense to add a custom character plus a flag, such as the hidden flag? For example, if I dot not want IGFD to list the files or directory that starts with a specific character (`!` for example ;-)), then I can by activating the flag and defining the character in the config header file. This is very specific, so I would totally understand if it is not the purpose of the library to do such a thing.

Anyway, thank you for your time, I really appreciate.

**Comment by aiekick** [comment]
i think i can add something for filter more the listing more specific for all dir or per dir. will think about it. 

because with regex you have a way for filter dir or file to show, but you dont have pattern to ignore and that can be some specific files or dirs like in your case.

for now yes we can close
